src/routes/popup/arns/setPrimaryName/ArNSConfirmSetPrimaryNameView.tsx

```diff
@@ -0,0 +1,128 @@
+import { ButtonV2 } from "@arconnect/components";
+import { Text } from "@arconnect/components-rebrand";
+import { useMemo, useState } from "react";
+import { Flex } from "~components/common/Flex";
+import HeadV2 from "~components/popup/HeadV2";
+import { ARNS_QUERY_CLIENT, setPrimaryName, useArioBalance, usePrimaryNameCostDetails, useTicker } from "~lib/arns";
+import { NAME_SERVICE_QUERY_CLIENT } from "~lib/nameservice";
+import { useActiveWallet } from "~wallets/hooks";
+import type { CommonRouteProps } from "~wallets/router/router.types";
+import { useLocation } from "~wallets/router/router.utils";
+import { CardContainer } from "../CardContainer";
+import TransactionStatusModal from "../TransactionStatusModal";
+import { sleep } from "~utils/promises/sleep";
+import { formatArio } from "../utils";
+import { mARIOToken } from "@ar.io/sdk/web";
+import { ProfileName } from "~components/ProfileName";
+
+export interface ArNSConfirmSetPrimaryNameViewParams {
+  name: string;
+}
+export type ArNSConfirmSetPrimaryNameViewProps = CommonRouteProps<ArNSConfirmSetPrimaryNameViewParams>;
+
+export const ArNSConfirmSetPrimaryNameView = ({ params: { name } }: ArNSConfirmSetPrimaryNameViewProps) => {
+  const [processingTransaction, setProcessingTransaction] = useState<boolean>(false);
+
+  const [transactionState, setTransactionState] = useState<string | undefined>();
+
+  const { data: costDetails } = usePrimaryNameCostDetails({ name });
+  const { data: ticker } = useTicker();
+
+  const costDetailsArio = useMemo(() => {
+    return costDetails ? new mARIOToken(costDetails.tokenCost).toARIO().valueOf() : undefined;
+  }, [costDetails]);
+  const formattedCost = useMemo(() => {
+    return costDetails ? formatArio(costDetailsArio) : undefined;
+  }, [costDetailsArio]);
+
+  const arioBalance = useArioBalance();
+
+  console.log("COST and Balance: ", costDetailsArio, arioBalance);
+
+  const wallet = useActiveWallet();
+
+  const { navigate } = useLocation();
+
+  const handleConfirmSetPrimaryName = async () => {
+    try {
+      setProcessingTransaction(true);
+      setTransactionState("Processing transactions...");
+
+      const result = await setPrimaryName({
+        name,
+        transactionListener: (transactionState) => {
+          console.log("transaction state received: ", transactionState);
+          setTransactionState(transactionState);
+        },
+      });
+
+      if (result.success) {
+        await sleep(2000);
```

Comment:
Seems a bit hacky. Might be worth adding a comment to clarify why that's needed.

Reply:
It is a bit hacky but it's there to account for AO's message cranking which has been a bit problematic. It generally has a slight delay and goes through, which this sleep covers well. In rare occasions the message doesn't get cranked at all and that isn't something that is easily handled nor detected.

Reply:
Added a comment to the code about the sleep

Reply:
https://github.com/wanderwallet/Wander/pull/1001/commits/2073c296cf0f8c380c02cb4f293d3a072caefa66